xarray/core/combine.py

```diff
@@ -323,16 +324,17 @@ def _dataarray_concat(arrays, dim, data_vars, coords, compat,
         raise ValueError('data_vars is not a valid argument when '
                          'concatenating DataArray objects')
 
-    datasets = []
-    for n, arr in enumerate(arrays):
-        if n == 0:
-            name = arr.name
-        elif name != arr.name:
-            if compat == 'identical':
-                raise ValueError('array names not identical')
-            else:
-                arr = arr.rename(name)
-        datasets.append(arr._to_temp_dataset())
+    name = result_name(arrays)
+    if name is None and compat == 'identical':
+        raise ValueError('array names not identical')
+    datasets = [arr.rename(name)._to_temp_dataset() for arr in arrays]
+
+    if name is None and isinstance(dim, str) and dim not in arrays[0].dims:
```

Comment:
This should verify that there are no coordinates with the dimension name, either — and probably  should check every argument, too. I think we support some cases with broadcasting where dimensions/coordinates can differ across arguments.

Reply:
We now check that the string `dim` is not in the coords or dims of any of the arrays to concatenate.